openshift_tools/monitoring/hawk_client.py

```diff
@@ -0,0 +1,117 @@
+    ml.append(new_metric)
+
+    # No Auth
+    connection = HawkConnection(url='172.17.0.27:8000')
+
+    # Basic Auth
+    connection = HawkConnection(url='172.17.0.27:8000', user='user', passwd='password')
+
+    client = HawkClient(connection)
+    print client.add_metric(ml)
+"""
+#These are not installed on the buildbot, disabling this
+#pylint: disable=no-name-in-module,unused-import
+from openshift_tools.monitoring.metricmanager import UniqueMetric, MetricManager
+from openshift_tools.monitoring.hawk_common import HawkConnection
+
+# The hawkular client is optional
+# If we only send data to zabbix we do not need/use the hawkular client
+try:
+    from hawkular.metrics import HawkularMetricsClient, MetricType, Availability
+except ImportError:
+    pass
+
+#This class implements rest calls. We only have one rest call implemented
+# add-metric.  More could be added here
+#pylint: disable=too-few-public-methods
+class HawkClient(object):
+    """
+    wrappers class around hawkular client python so use can use it with UniqueMetric
+    """
+
+    def __init__(self, hawk_connection, headers=None):
+        # pylint doesn't know where RestAPI is
+        #pylint: disable=undefined-variable
+        self.hawk_conn = hawk_connection
+        self.client = None
+
+        # Do not create a client if inactive
+        if self.hawk_conn.inactive:
+            return
+
+        self.client = HawkularMetricsClient(host=self.hawk_conn.host,
+                                            port=self.hawk_conn.port,
+                                            scheme=self.hawk_conn.scheme,
+                                            username=self.hawk_conn.username,
+                                            password=self.hawk_conn.password,
+                                            context=self.hawk_conn.context,
+                                            tenant_id=self.hawk_conn.tenant_id,
+                                           )
+
+    def push_heartbeat(self, metric):
+        """
+        Push a list of heartbeat metric-endpoints via hawkular client
+        """
+        # We assume the value of a heartbeat is a dict
+        heartbeat = metric.value
+        # Hawkular metrics heartbeat value is "up"
+        value = Availability.Up
+        # Hawkular metrics use milliseconds
+        clock = metric.clock * 1000
+        # Use MetricType.Availability for heartbeat data
+        metric_type = MetricType.Availability
+
+        for template in heartbeat.get('templates'):
+            # Add the group and host to the key
+            key = '{0}/{1}/{2}'.format('ops/heartbeat', 'template', template)
+            self.client.push(metric_type, key, value, clock)
+
+        for hostgroup in heartbeat.get('hostgroups'):
+            # Add the group and host to the key
+            key = '{0}/{1}/{2}'.format('ops/heartbeat', 'hostgroup', hostgroup)
+            self.client.push(metric_type, key, value, clock)
```

Comment:
As I understand it In our end goal, instead of the host groups we would have each metric connected to an entity we know about: Node/Container/etc. 

With regards to hawkular I'm not sure if it makes more sense to concat keys or to use tags. Bare in mind we might have more then one key.